File: C++/238. Product of Array Except Self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& n) {
        int z=0,t;
        long long int s=1;
        for(int i=0;i<n.size();i++){
            if(n[i]==0){
            z++;
            t=i;}
            else
            s*=n[i];
        }
        if(z>1){
            vector<int>a(n.size(),0);
            return a;
        }
        else if(z==1){
            vector<int>a(n.size(),0);
            a[t]=s;
            return a;
        }
        for(int i=0;i<n.size();i++){
            if(n[i]==0)
            n[i]=s;
            else
            n[i]=(s/n[i]);
        }
        return n;
    }
};

File: JAVA/238. Product of Array Except Self.java
class Solution {
    public int[] productExceptSelf(int[] n) {
        int z = 0, t = -1;
        long s = 1;

        for (int i = 0; i < n.length; i++) {
            if (n[i] == 0) {
                z++;
                t = i;
            } else {
                s *= n[i];
            }
        }

        int[] a = new int[n.length];

        if (z > 1) {
            return a;
        }
        
        if (z == 1) {
            a[t] = (int) s;
            return a;
        }

        for (int i = 0; i < n.length; i++) {
            a[i] = (int) (s / n[i]);
        }

        return a;
    }
}